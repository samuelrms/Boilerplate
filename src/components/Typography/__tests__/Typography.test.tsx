import { renderWithTheme } from '@/tests/utils';
import { screen } from '@testing-library/react';

import { Typography } from '..';

describe('Typography', () => {
  test('renders', () => {
    renderWithTheme(<Typography />);
    const typography = screen.getByTestId('typography');
    expect(typography);
  });

  test('should render a text', () => {
    const { getByText } = renderWithTheme(<Typography>Typography</Typography>);
    const typography = getByText('Typography');
    expect(typography).toBeInTheDocument();
  });

  test('should render with custom color', () => {
    renderWithTheme(<Typography color="red" />);
    const typography = screen.getByTestId('typography');
    expect(typography).toHaveStyle({ color: 'red' });
  });

  test('should render with custom size', () => {
    renderWithTheme(<Typography size="sm" />);
    const typography = screen.getByTestId('typography');
    expect(typography).toHaveStyle({ fontSize: '0.875rem' });
  });

  test('should render with custom variant', () => {
    renderWithTheme(<Typography variant="caption" />);
    const typography = screen.getByTestId('typography');
    expect(typography).toHaveStyle({ fontSize: '0.75rem' });
  });

  test('should render with custom size and variant', () => {
    renderWithTheme(<Typography size="sm" variant="caption" />);
    const typography = screen.getByTestId('typography');
    expect(typography).toHaveStyle({ fontSize: '0.875rem' });
  });

  // test('should render correctly', () => {
  //   const { container } = renderWithTheme(<Typography></>);
  //   expect(container.firstChild).toMatchSnapshot();
  // });
});
