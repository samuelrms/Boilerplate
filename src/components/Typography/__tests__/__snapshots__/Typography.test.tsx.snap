// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Typography should render correctly 1`] = `
<p
  class="css-1ndg5q"
  data-testid="typography"
>
  Typography
</p>
`;
